Reply/app/src/main/java/com/example/reply/ui/ReplyListContent.kt

```diff
@@ -0,0 +1,185 @@
+import com.example.reply.ui.components.EmailDetailAppBar
+import com.example.reply.ui.components.ReplyEmailListItem
+import com.example.reply.ui.components.ReplyEmailThreadItem
+import com.example.reply.ui.components.ReplySearchBar
+import com.example.reply.ui.utils.ReplyContentType
+import com.example.reply.ui.utils.ReplyNavigationType
+
+@Composable
+fun ReplyInboxScreen(
+    contentType: ReplyContentType,
+    replyHomeUIState: ReplyHomeUIState,
+    navigationType: ReplyNavigationType,
+    closeDetailScreen: () -> Unit,
+    navigateToDetail: (Long, ReplyContentType) -> Unit
+) {
+    /**
+     * When moving from LIST page to LIST_AND_DETAIL page on screen expand, set first email selected as default
+     * When moving from LIST_AND_DETAIL page to LIST page clear the selection and user should see LIST screen.
+     */
+
+    LaunchedEffect(key1 = contentType) {
+        if (contentType == ReplyContentType.DUAL_PANE && replyHomeUIState.selectedEmail == null) {
+            replyHomeUIState.emails.firstOrNull()?.let { firstEmail ->
+                navigateToDetail.invoke(firstEmail.id, ReplyContentType.DUAL_PANE)
+            }
+        } else if (contentType == ReplyContentType.SINGLE_PANE && !replyHomeUIState.showDetailScreenOnly) {
+            closeDetailScreen.invoke()
+        }
+    }
+
+    val emailLazyListState = rememberLazyListState()
+
+    if (contentType == ReplyContentType.DUAL_PANE) {
+        ReplyListAndDetailContent(
+            replyHomeUIState = replyHomeUIState,
+            emailLazyListState = emailLazyListState,
+            modifier = Modifier.fillMaxSize(),
+            navigateToDetail = navigateToDetail
+        )
+    } else {
+        Box(modifier = Modifier.fillMaxSize()) {
+            ReplyListOnlyContent(
+                replyHomeUIState = replyHomeUIState,
+                emailLazyListState = emailLazyListState,
+                modifier = Modifier.fillMaxSize(),
+                closeDetailScreen = closeDetailScreen,
+                navigateToDetail = navigateToDetail
+            )
+            // When we have bottom navigation we show FAB at the bottom end.
+            if (navigationType == ReplyNavigationType.BOTTOM_NAVIGATION) {
+                LargeFloatingActionButton(
+                    onClick = { /*TODO*/ },
+                    modifier = Modifier
+                        .align(Alignment.BottomEnd)
+                        .padding(16.dp),
+                    containerColor = MaterialTheme.colorScheme.tertiaryContainer,
+                    contentColor = MaterialTheme.colorScheme.onTertiaryContainer
+                ) {
+                    Icon(
+                        imageVector = Icons.Default.Edit,
+                        contentDescription = stringResource(id = R.string.edit),
+                        modifier = Modifier.size(28.dp)
+                    )
+                }
+            }
+        }
+    }
+}
+
+@Composable
+fun ReplyListOnlyContent(
```

Comment:
Nit: This is still `ListOnly`, even though it might show either the detail or the list.

Reply:
Updated to `ReplySinglePaneContent` 👍🏽